tests/kernel/boot_page_table/src/boot_page_table.c

```diff
@@ -0,0 +1,361 @@
+		value = X86_MMU_GET_PTE(addr_range);
+		status &= check_param_nonset_region(value,
+						     REGION_PERM);
+		if (status == 0) {
+			TC_PRINT("%s failed\n", __func__);
+			printk("error at %d pte %d permissions %d\n",
+				addr_range, value.value, REGION_PERM);
+			return TC_FAIL;
+
+		}
+	}
+	return TC_PASS;
+}
+
+static int ending_start_addr_range2(void)
+{
+
+	u32_t addr_range, status = true;
+	union x86_mmu_pte value;
+
+	for (addr_range = START_ADDR_RANGE2 + ADDR_SIZE; addr_range <
+	     (START_ADDR_RANGE2 + ADDR_SIZE + 0x10000);
+	       addr_range += 0x1000) {
+		value = X86_MMU_GET_PTE(addr_range);
+		status &= check_param_nonset_region(value, REGION_PERM);
+		if (status == 0) {
+			TC_PRINT("%s failed\n", __func__);
+			printk("error at %d pte %d permissions %d\n",
+				addr_range, value.value, REGION_PERM);
+			return TC_FAIL;
+
+		}
+	}
+	return TC_PASS;
+}
+
+/* START_ADDR_RANGE3 */
+static int  starting_addr_range3(void)
+{
+
+	u32_t addr_range, status = true;
+	union x86_mmu_pte value;
+
+	for (addr_range = START_ADDR_RANGE3; addr_range <=
+	     (START_ADDR_RANGE3 + STARTING_ADDR_RANGE_LMT);
+	       addr_range += 0x1000) {
+		value = X86_MMU_GET_PTE(addr_range);
+		status &= check_param(value, REGION_PERM);
+		if (status == 0) {
+			printk("error at %d pte %d permissions %d\n",
+				addr_range, value.value, REGION_PERM);
+			TC_PRINT("%s failed\n", __func__);
+			return TC_FAIL;
+		}
+	}
+	return TC_PASS;
+}
+
+static int before_start_addr_range3(void)
+{
+
+	u32_t addr_range, status = true;
+	union x86_mmu_pte value;
+
+	for (addr_range = START_ADDR_RANGE3 - 0x7000;
+	     addr_range < (START_ADDR_RANGE3); addr_range += 0x1000) {
+		value = X86_MMU_GET_PTE(addr_range);
+		status &= check_param_nonset_region(value,
+						     REGION_PERM);
+		if (status == 0) {
+			TC_PRINT("%s failed\n", __func__);
+			printk("error at %d pte %d permissions %d\n",
+				addr_range, value.value, REGION_PERM);
+			return TC_FAIL;
+
+		}
+	}
+	return TC_PASS;
+}
+
+static int ending_start_addr_range3(void)
+{
+
+	u32_t addr_range, status = true;
+	union x86_mmu_pte value;
+
+	for (addr_range = START_ADDR_RANGE3 + ADDR_SIZE; addr_range <
+	     (START_ADDR_RANGE3 + ADDR_SIZE + 0x10000);
+	       addr_range += 0x1000) {
+		value = X86_MMU_GET_PTE(addr_range);
+		status &= check_param_nonset_region(value,
+						     REGION_PERM);
+		if (status == 0) {
+			TC_PRINT("%s failed\n", __func__);
+			printk("error at %d pte %d permissions %d\n",
+				addr_range, value.value, REGION_PERM);
+			return TC_FAIL;
+
+		}
+	}
+	return TC_PASS;
+}
+void test_starting_addr_range(void)
+{
+	zassert_true(starting_addr_range() == TC_PASS, NULL);
+}
+
+void test_before_start_addr_range(void)
+{
+	zassert_true(before_start_addr_range() == TC_PASS, NULL);
+}
+
+void test_ending_start_addr_range(void)
+{
+	zassert_true(ending_start_addr_range() == TC_PASS, NULL);
+}
+
+void test_starting_addr_range1(void)
```

Comment:
Pass arguments into the same function. if arguments== 1 then take start addr 1 if argument == 2 then take start addr2 so on.
Don't duplicate the code.

Reply:
@AdithyaBaglody . The memory address are already passed based on the arguments.